### PR TITLE
robot type chooser -> master

### PR DESCRIPTION
**Pull request by greenblitz4590**

### CODE REVIEW

**Review by yonnbro — APPROVED** [review]

**Comment by yonnbro** [comment]
We need to make sure the code works on a real robot because it could not work when robot code starts but not connected to a driver station yet

File: src/main/java/edu/greenblitz/robotName/Robot.java [hunk 1]
```diff
@@ -123,4 +124,18 @@ private void initializeLogger() {
         }
         Logger.start();
     }
+    public static RobotType getRobotType (){
+        RobotType robotType = RobotConstants.ROBOT_TYPE;
+        if(isReal()){
```
Comment (yonnbro) [inline src/main/java/edu/greenblitz/robotName/Robot.java:129]: What is “isReal” ?

Comment (yonnbro) [reply src/main/java/edu/greenblitz/robotName/Robot.java:129]: Where is the code located?

Comment (NoamRosenberg08) [reply src/main/java/edu/greenblitz/robotName/Robot.java:129]: in the parent class of the Robot.java class

File: src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterFactory.java [hunk 2]
```diff
@@ -8,10 +8,10 @@
 
 public class LifterFactory {
     public static ILifter create() {
-        return switch (RobotConstants.ROBOT_TYPE) {
+        return switch (Robot.getRobotType()) {
             case ROBOT_NAME -> new NeoLifter();
             case REPLAY -> new ReplayLifter();
-            case SIMULATION -> new SimulationLifter();
+            default-> new SimulationLifter();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/Lifter/LifterFactory.java:14]:
```suggestion
            case SIMULATION, PEGA_SWERVE-> new SimulationLifter();
```

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/ElbowFactory.java [hunk 3]
```diff
@@ -1,15 +1,16 @@
...
 import edu.greenblitz.robotName.subsystems.arm.elbow.SimulationElbow.SimulationElbow;
 
 public class ElbowFactory {
     public static IElbow create() {
-        return switch (RobotConstants.ROBOT_TYPE) {
+        return switch (Robot.getRobotType()) {
             case ROBOT_NAME -> new NeoElbow();
             case REPLAY -> new ReplayElbow();
-            case SIMULATION -> new SimulationElbow();
+            default -> new SimulationElbow();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/elbow/ElbowFactory.java:13]:
```suggestion
            case SIMULATION, PEGA_SWERVE -> new SimulationElbow();
```

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/roller/RollerFactory.java [hunk 4]
```diff
@@ -1,15 +1,16 @@
...
 import edu.greenblitz.robotName.subsystems.arm.roller.SimulationRoller.SimulationRoller;
 
 public class RollerFactory {
 	public static IRoller create() {
-		return switch (RobotConstants.ROBOT_TYPE) {
+		return switch (Robot.getRobotType()) {
 			case ROBOT_NAME -> new BagRoller();
 			case REPLAY -> new ReplayRoller();
-			case SIMULATION -> new SimulationRoller();
+			default -> new SimulationRoller();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/roller/RollerFactory.java:13]:
```suggestion
			case SIMULATION, PEGA_SWERVE -> new SimulationRoller();
```

File: src/main/java/edu/greenblitz/robotName/subsystems/arm/wrist/WristFactory.java [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 import edu.greenblitz.robotName.subsystems.arm.wrist.SimulationWrist.SimulationWrist;
 
 public class WristFactory {
     public static IWrist create() {
-        return switch (RobotConstants.ROBOT_TYPE) {
+        return switch (Robot.getRobotType()) {
             case ROBOT_NAME -> new NeoWrist();
             case REPLAY -> new ReplayWrist();
-            case SIMULATION -> new SimulationWrist();
+            default -> new SimulationWrist();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/arm/wrist/WristFactory.java:13]:
```suggestion
            case SIMULATION, PEGA_SWERVE -> new SimulationWrist();
```

File: src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/PivotFactory.java [hunk 6]
```diff
@@ -1,16 +1,17 @@
...
 
 
 public class PivotFactory {
     public static IPivot create() {
-        return switch (RobotConstants.ROBOT_TYPE) {
+        return switch (Robot.getRobotType()) {
             case ROBOT_NAME -> new FalconPivot();
-            case SIMULATION -> new SimulationPivot();
             case REPLAY -> new ReplayPivot();
+            default -> new SimulationPivot();
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/shooter/Pivot/PivotFactory.java:14]:
```suggestion
             case SIMULATION, PEGA_SWERVE-> new SimulationPivot();
```

File: src/main/java/edu/greenblitz/robotName/subsystems/swerve/Modules/SwerveModuleFactory.java [hunk 7]
```diff
@@ -10,10 +11,11 @@
 public class SwerveModuleFactory {
 
     public static ISwerveModule create(SwerveChassis.Module module) {
-        return switch (RobotConstants.ROBOT_TYPE) {
+        return switch (Robot.getRobotType()) {
             case REPLAY -> new ReplaySwerveModule();
             case ROBOT_NAME -> new MK4ISwerveModule(module);
-            case SIMULATION -> new SimulationSwerveModule(module);
+            case PEGA_SWERVE -> new KazaSwerveModule(module);
+			default -> new SimulationSwerveModule(module);
```
Comment (bossyshrimp68) [inline src/main/java/edu/greenblitz/robotName/subsystems/swerve/Modules/SwerveModuleFactory.java:18]:
```suggestion
			case SIMULATION -> new SimulationSwerveModule(module);
```

**Review by bossyshrimp68 — CHANGES_REQUESTED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Pull request merged**